File: dsa-problems/Queues/solutions/java/QUE-003-cafeteria-queue-rotation.java
import java.util.*;

class Solution {
    public String processQueueOperations(List<String[]> operations) {
        Queue<Integer> q = new LinkedList<>();
        long total = 0;

        for (String[] opData : operations) {
            String cmd = opData[0];

            if (cmd.equals("ENQUEUE")) {
                q.offer(Integer.parseInt(opData[1]));
            } else if (cmd.equals("DEQUEUE")) {
                if (!q.isEmpty()) {
                    q.poll();
                }
            } else if (cmd.equals("FRONT")) {
                // Just read
            } else if (cmd.equals("REAR")) {
                // Just read
            } else if (cmd.equals("SIZE")) {
                total += q.size();
            } else if (cmd.equals("ISEMPTY")) {
                // Just read
            }
        }

        return String.valueOf(total);
    }
}

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        if (sc.hasNextInt()) {
            int m = sc.nextInt();
            List<String[]> operations = new ArrayList<>();

            for (int i = 0; i < m; i++) {
                String op = sc.next();
                if (op.equals("ENQUEUE")) {
                    String val = sc.next();
                    operations.add(new String[]{op, val});
                } else {
                    operations.add(new String[]{op});
                }
            }

            Solution solution = new Solution();
            String result = solution.processQueueOperations(operations);
            System.out.println(result);
        }
        sc.close();
    }
}

File: dsa-problems/Queues/solutions/javascript/QUE-003-cafeteria-queue-rotation.js
const readline = require("readline");

class Solution {
  processQueueOperations(operations) {
    const queue = [];
    let total = 0;

    for (const opData of operations) {
      const cmd = opData[0];

      if (cmd === "ENQUEUE") {
        queue.push(parseInt(opData[1], 10));

      } else if (cmd === "DEQUEUE") {
        if (queue.length > 0) {
          queue.shift();
        }

      } else if (cmd === "FRONT") {
        // Just read
      } else if (cmd === "REAR") {
        // Just read
      } else if (cmd === "SIZE") {
        total += queue.length;

      } else if (cmd === "ISEMPTY") {
        // Just read
      }
    }

    return String(total);
  }
}

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let data = [];
rl.on("line", (line) => data.push(...line.trim().split(/\s+/).filter(x => x !== "")));
rl.on("close", () => {
  if (data.length === 0) return;
  let idx = 0;
  const m = parseInt(data[idx++], 10);
  const operations = [];

  for (let i = 0; i < m; i++) {
    const op = data[idx++];
    if (op === "ENQUEUE") {
      const val = data[idx++];
      operations.push([op, val]);
    } else {
      operations.push([op]);
    }
  }

  const solution = new Solution();
  const result = solution.processQueueOperations(operations);
  console.log(result);
});
